src/mds/MDSRank.cc

```diff
@@ -2384,6 +2384,18 @@ void MDSRankDispatcher::handle_mds_map(
   if (mdsmap->get_inline_data_enabled() && !oldmap.get_inline_data_enabled())
     dout(0) << "WARNING: inline_data support has been deprecated and will be removed in a future release" << dendl;
 
+  if (scrubstack->is_scrubbing() && mdsmap->get_max_mds() > 1) {
+    std::lock_guard l(mds_lock);
```

Comment:
This method already executes with the `mds_lock` locked. See `MDSDaemon::ms_dispatch2`.

Reply:
> This method already executes with the `mds_lock` locked. See `MDSDaemon::ms_dispatch2`.

removed the locking